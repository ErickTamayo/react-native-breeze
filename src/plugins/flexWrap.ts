import { PluginFunction } from "./types";

export type PluginGroups = {
  wrap: "wrap" | "no-wrap" | "wrap-reverse";
};

export const pattern = /^flex-(?<wrap>wrap|no-wrap|wrap-reverse)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { wrap } = groups;
  const flexWrap = wrap === "no-wrap" ? "nowrap" : wrap;
  return { flexWrap };
};
