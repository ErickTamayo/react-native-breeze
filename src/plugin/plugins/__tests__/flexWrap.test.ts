import { pattern, plugin } from "../flexWrap";
import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";

describe("flexWrap", () => {
  // prettier-ignore
  const shouldMatch = generateInput("flex", [ "wrap", "no-wrap", "wrap-reverse"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown", "flex-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
