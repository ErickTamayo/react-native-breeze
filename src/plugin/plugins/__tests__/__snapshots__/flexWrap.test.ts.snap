// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flexWrap Should parse (flex-no-wrap) correctly 1`] = `
Object {
  "flexWrap": "nowrap",
}
`;

exports[`flexWrap Should parse (flex-wrap) correctly 1`] = `
Object {
  "flexWrap": "wrap",
}
`;

exports[`flexWrap Should parse (flex-wrap-reverse) correctly 1`] = `
Object {
  "flexWrap": "wrap-reverse",
}
`;
